### ISSUE TITLE
Spec incorrectly states *.example.com “matches any resource on the host”?

### ISSUE THREAD

**Issue by sideshowbarker** [w3c/webappsec-csp#241]
A part of [the definition of **source expression** in the spec](https://w3c.github.io/webappsec-csp/#source-expression) appears to be wrong; this (emphasis added):

> *.example.com (which _matches any resource on the host_ or any of its subdomains

That seems wrong because, given the following policy:

    Content-Security-Policy: script-src self https://*.example.com

…then that spec language seems to be saying the above policy — along with allowing scripts from any https subdomains of example.com — will allow scripts from `https://example.com` itself.

But that doesn’t seem to be true. The relevant normative requirements for UAs at https://w3c.github.io/webappsec-csp/#match-hosts say this:

> 1. If the first character of _A_ is an U+002A ASTERISK character (*):
>
>> 1. Let _remaining_ be the result of removing the leading ("`*`") from _A_.
>> 2. If _remaining_ (including the leading U+002E FULL STOP character (`.`)) is an ASCII case-insensitive match for the rightmost characters of _B_, then return "`Matches`". Otherwise, return "`Does Not Match`".

The *including the leading U+002E FULL STOP character (`.`)* part of the requirement indicates the *remaining* part of *A* after the asterisk is removed includes the leading full stop, and so the rightmost characters of *B* must also start with a dot in order to match that.

In other words, if you start with `*.example.com` as *A* and walk through that part of the algorithm, you start by removing the `*` to get `.example.com` as the *remaining* part, and then you match the rightmost characters of *B* against that, including the leading full stop.

So `https://foo.example.com ` matches, because the rightmost characters of its host part match `.example.com`, but `https://example.com` doesn’t match, because the rightmost characters of its host part don’t match `.example.com` (because it lacks the included full stop).

See also https://stackoverflow.com/a/44857848/441757

**Comment by andypaicu** [comment]
Thank you @sideshowbarker for reporting this, I've updated the spec.
